magma/passes/stub.py

```diff
@@ -0,0 +1,30 @@
+from typing import Iterable
+
+from magma.circuit import CircuitKind
+from magma.passes.passes import CircuitPass
+
+
+def _drive_if_undriven_input(port):
+    if port.is_mixed():
+        for p in port:
+            _drive_if_undriven_input(p)
+        return
+    # NOTE(rsetaluri): This may override previously wired inputs, resulting in a
+    # warning. In this case, this warning is benign.
+    if port.is_input():
+        port.undriven()
```

Comment:
Perhaps we should disable warnings in this pass? One simple way would be to set a magma "global" that disables warning reporting, which we can enable/disable in this pass.  A more fine-grained way might be to pass a flag through undriven that says ignore drivers or something.  Ideally we can avoid generating warnings internally, even if they are benign